bundler/spec/bundler/cli_spec.rb

```diff
@@ -161,16 +161,16 @@ def out_with_macos_man_workaround
       it "prints json output data when there are outdated gems" do
         run_command
         out_data = JSON.parse(out)
-        expect(out_data.keys).to contain_exactly("outdated_count", "outdated_gems")
-        expect(out_data["outdated_count"]).to eq(1)
-        expect(out_data["outdated_gems"].length).to eq(1)
-
-        gem_data = out_data["outdated_gems"].first
-        expect(gem_data).to include({
-          "current_spec" => hash_including("name" => "rack", "version" => "0.9.1"),
-          "active_spec" => hash_including("name" => "rack", "version" => "1.0.0"),
-          "dependency" => "rack (= 0.9.1)",
-          "groups" => ["default"],
+        expect(out_data.length).to eq(1)
+
+        expect(out_data.first).to eq({
+          "gem_name" => "rack",
+          "current_version" => "0.9.1",
+          "latest_version" => "1.0.0",
+          "latest_required_ruby" => ">= 0",
+          "latest_required_rubygems" => ">= 0",
+          "requested_version" => "= 0.9.1",
+          "groups" => "default",
```

Comment:
What's this do if there's more than 1 group? There should at least be a test.

I'm guessing we lose the array because of the to_s, and I'm thinking it gets concatenated. Would it be better to allow whatever comes in for types and expect the caller to convert types to string if necessary?

Reply:
So, the original presentation glues them together with commas, and my initial `--json` implementation split them back apart. The way it would have to work if we want them to be an array in the json output is that the `Shell#table` implementation would need to detect that it got an array, and `map(&:to_s).join(", ")` them, which .. isn't awful, but I wasn't sure was really generalizable enough.

I think I'd like it a little better though, and if you're also game I'll go ahead and add it :-)

Reply:
Added, and it cleaned up the handling of `groups` in Cli::Outdated a bit overall.

Reply:
Ah ok, I thought it received an array already so I was incorrect in my review. I see it's not so simple now because of how Shell handles it. 

So the choice is between preformatted csv in the json or detecting the array in Shell.

I agree with detecting the array.